### PR TITLE
 Docs for Devbox Run and Setting the Commit Sha

### PR DESCRIPTION
**Pull request by Lagoja**
## Summary

Docs for Devbox Run and Setting the Commit Sha

## How was it tested?

localhost

### CODE REVIEW

File: docs/app/docs/guides/scripts.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Again
+```
+
+Your devbox shell will exit once the last line of your script has finished running, or when you interrupt the script with CTRL-C (or a SIGINT signal).
+
+
+## Tips on using Scripts
+
+1. Since `init_hook` runs everytime you start your shell, you should use primarily use it for setting environment variables and aliases. For longer running tasks like database setup, you can create and run a Devbox script
+2. You can use Devbox scripts to start and manage long running background processes and daemons. For example -- If you are working on a LAMP stack project, you can use scripts to start Mysql and Apache in separate shells and monitor their logs. Once you are done developing, you can use CTRL-C to exit the processes and shells
```
Comment (ipince) [inline docs/app/docs/guides/scripts.md:51]: Mysql -> MySQL?

Comment (ipince) [reply docs/app/docs/guides/scripts.md:51]: Also, we used "Ctrl-C" above but "CTRL-C" here. We should pick one. I think the code has "CTRL-C" (but of course we can change that).

File: docs/app/docs/guides/scripts.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+3. If a script feels too long to put it directly in `devbox.json`, you can save it as a shell script in your project, and then invoke it in your `devbox scripts`.
+4. For more ideas, see the LAMP stack example in our Devbox examples repo.
```
Comment (ipince) [inline docs/app/docs/guides/scripts.md:53]: link to it?

**Review by ipince — APPROVED** [review]
LGTM on the scripts front

File: docs/app/docs/configuration.md [hunk 3]
```diff
@@ -53,6 +63,38 @@ Welcome! See CONTRIBUTING.md for tips on contributing to devbox.
...
+        "echo \"Hello Twice!\""
+    ]
+}
+```
+
+### Nixpkgs
+
+The Nixpkg object is used to optionally configure which version of the Nixpkgs repository you want Devbox to use for installing packages. It currently takes a single field, `commit`, which takes a commit hash for the specific revision of Nixpkgs you want to use.
+
+If a Nixpkg commit is not set, Devbox will automatically add a default commit hash to your `devbox.json`. To upgrade your packages tp the latest available versions in the future, you can replace the default hash with the latest nixpkgs-unstable hash from https://status.nixos.org
```
Comment (savil) [inline docs/app/docs/configuration.md:93]: > To upgrade your packages tp the latest

typo: to

File: docs/app/docs/guides/pinning_packages.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+```json
+"nixpkgs": {
+    "commit": "89f196fe781c53cb50fef61d3063fa5e8d61b6e5"
+}
+```
+This hash ensures that Devbox will install the same packages whenever you start a shell. By checking this into source control, you can also ensure that any other developers who run your project will get the same packages.
+
+## Using the latest version of Nixpkgs
+
+To use the latest available packages in Nix, you can replace the commit in devbox.json with the latest `nixpkgs-unstable` hash from https://status.nixos.org.
```
Comment (savil) [inline docs/app/docs/guides/pinning_packages.md:26]: tildes needed around `devbox.json`

File: docs/app/docs/guides/pinning_packages.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+```
+This hash ensures that Devbox will install the same packages whenever you start a shell. By checking this into source control, you can also ensure that any other developers who run your project will get the same packages.
+
+## Using the latest version of Nixpkgs
+
+To use the latest available packages in Nix, you can replace the commit in devbox.json with the latest `nixpkgs-unstable` hash from https://status.nixos.org. 
+
+## Look up a commit hash for a specific package
+
+In most cases, the packages available in Devbox's default commit should suffice for your usecases. However, if you need a specific minor version, or an older version of a package that is no longer included in Nixpkgs, you may need update the commit SHA Unfortunately, Nix does not have an official way to find the Nixpkg commit SHA for a specific version of a package.
```
Comment (savil) [inline docs/app/docs/guides/pinning_packages.md:30]: period before "Unfortunately"

**Review by savil — APPROVED** [review]

**Pull request merged**